Merging insert_column.py and test_insert_column.py

ShivaShinde opened the pull request:
I had to insert columns to my spreadsheet and test it. Since Gspread has no insert_column method, I have written this script, which inserts new columns at any given index. Thanks!!

ShivaShinde commented:
@burnash requesting for a merge!!

Hi, please check my insert_column.py & test_insert_column.py. I tested these methods, I am using it, they are working perfectly fine. There is anything should be done before requesting for a merge here?

(I was not familiar with the github before, So by mistake I did all my code changes on github (forked project). Made 31 commits, which is so many. But I didn't edit any other method, I am sure about this. All my commits are on insert_column.py and test_insert_column.py. I tested this, which is running perfectly fine.)

burnash commented on gspread/models.py:
```diff
@@ -10,6 +10,7 @@
 
 
 import re
+import pdb
```
Please remove unused module

ShivaShinde replied:
I removed this.

burnash commented on gspread/models.py:
```diff
@@ -563,6 +564,36 @@ def insert_row(self, values, index=1):
+            self.resize(rows=data_width)
+	
+	all_cells = self.get_all_values()
+        j=0
```
This looks like falling out of indentation

ShivaShinde replied:
I guess this is not one of the recent commits. Anyways, thanks for checking this, I have re-checked and I don't see any indentation problems.

burnash commented on gspread/models.py:
```diff
@@ -563,6 +564,36 @@ def insert_row(self, values, index=1):
+	    columns_after_insert = [[]]
+	else:
+	    columns_before_insert = [[] for k in range(len(all_cells[0]))]
+            while j<len(all_cells[0]):
```
General note: consider adhering to [PEP8](https://www.python.org/dev/peps/pep-0008/).

ShivaShinde replied:
I have corrected all the indentations. Please take a look into the latest commit.

burnash commented on gspread/models.py:
```diff
@@ -563,6 +564,36 @@ def insert_row(self, values, index=1):
         self.update_cells(updated_cell_list)
+    
+
+    def insert_column(self, values, index=1):
```
Add a [docstring](https://www.python.org/dev/peps/pep-0257/). Take [`insert_row`'s](https://github.com/burnash/gspread/blob/c0398ec19f129d18ed0e37abaee2481188bb7051/gspread/models.py#L546) as an example.

ShivaShinde commented:
@burnash Hi, I am confused, why the builds are failing. I saw build showing errors in test-config.py and some other files.

Please let me know if there is anything wrong in insert_column.py. This is my last and most recent commit: https://github.com/ShivaShinde/gspread/commit/8f67bf5cfb945f4045ee8924d4b5692ca4c852ff

burnash commented:
@ShivaShinde don't worry about the builds. I haven't finished setting up Travis yet, that's why they've been failing.

ShivaShinde commented:
@burnash Thanks for the update. Hoping that, from my last commit, there will be no errors anymore. Please merge this, when you finish all the setups.

ShivaShinde commented:
Hi @burnash  

Since its not merged yet, please let me know if there is anything to be done from my side. Thanks

ShivaShinde commented:
Hi @burnash  updated this branch. Please check. Thanks!!

ShivaShinde commented:
Hi, @msuozzo

Please help, I am new to github, Instead of cloning the repo and working on it. I was working on my forked repo on github, commiting for any modifications directly on github. Recently, I have known this is really a bad practise. Apology for that. 

I have written **insert_coulmn.py** and **test_insert_column.py** methods for my own project work. It worked perfectly for me, please check. So, I thought I should contribute here and I did, see this pull request. 

**Now the issue here is**, is the author of gspread facing any issues or conflicts while merging this branch. Am I missing something very important? 

Thanks for your time, I really appreciate it.

burnash commented:
Hi @ShivaShinde. I'm sorry for the silence, the issue I'm facing is a lack of time. 

I took a look at your [final commit](https://github.com/ShivaShinde/gspread/commit/fd97e5552ef0ab86ecc42a798252b8780d90495e).

First, you need to remove the `build` folder from git. It's a derived folder and there's no need for it in git. Second, you need to squash all of your commits into a single commit. That way it'll be much easier to review the PR and it won't litter the git log. See the detailed instruction on how to do squashing [here](https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request).

ShivaShinde commented:
Hi @burnash I didn't notice the build folder, now I removed it. I squashed all the commits into 1 single commit [this](https://github.com/ShivaShinde/gspread/commit/120bad77928849e98586661669f3e34dd9b9240b). I have also updated my branch on this [commit](https://github.com/ShivaShinde/gspread/commit/ec2e81b83015076f2bf9d81b1d7b1651060b8828).

Hope everything is all set. Please let me know if I am missing something. Thanks, I couldn't have done my project sooner without GSpread and thanks for all your suggestions.

ShivaShinde commented:
Hello @burnash and @msuozzo It's been more than an year, I had to clear all the merge conflicts (squashed them into 2 commits), now the branch is fresh and ready. Could you merge this pull request now?

Thanks 👍

ShivaShinde commented:
Hello @burnash once again I have squashed all the commits, now this branch is up to date, please check the changes :)

burnash commented:
Hi @ShivaShinde, sorry for the delayed reply and thank you for the contribution. Unfortunately, I don't have enough time to assist you with fixing [that many merge conflicts](https://github.com/burnash/gspread/pull/275/files#diff-32d879761dfc2378af26ee09994d1f3a). I've also released a new version of gspread that uses new API v4. The API v3 code is still around but will be eventually removed since Google intends to deprecate v3. I don't think it's practical to work on v3 related features anymore, so I'm closing this PR.
